src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -2267,6 +2271,81 @@ export class ResultsViewPageStore {
         }
     });
 
+    readonly referenceGenes = remoteData<ReferenceGenomeGene[]>({
+        await: ()=>[
+            this.studies
+        ],
+        invoke: () => {
+            return fetchAllReferenceGenomeGenes(this.studies.result[0].referenceGenome);
+        }
+    });
+
+    readonly hugoGeneSymbolToReferenceGene = remoteData({
+        await: ()=>[
+            this.referenceGenes
+        ],
+        invoke: ()=>{
+            // build reference gene map
+            const result:{[hugosymbol:string]:ReferenceGenomeGene} =
```

Comment:
theres no need to save `result`, just return `Promise.resolve(_.keyBy(...))`

if there's typing issues, you can annotate the type above:
`remoteData<{[hugoSymbol:string]:ReferenceGenomeGene}>({ ...`